src/ConnectionConfig.php

```diff
@@ -201,20 +153,20 @@ public function getCollation(): string
         return $this->collate;
     }
 
-    public function withCharset(string $charset, string $collate)
+    public function withCharset(string $charset = self::DEFAULT_CHARSET, string $collate = self::DEFAULT_COLLATE): self
```

Comment:
Not sure whether default parameters make sense here.

Reply:
At least the second is good, so I can set charset without passing new collate. But we can also remove it, it you prefer.